contrib/golang/filters/http/source/common/dso/dso.cc

```diff
@@ -0,0 +1,161 @@
+#include "contrib/golang/filters/http/source/common/dso/dso.h"
+
+#include "source/common/common/assert.h"
+
+namespace Envoy {
+namespace Dso {
+
+std::map<std::string, DsoInstance*> DsoInstanceManager::dso_map_ = {};
+absl::Mutex DsoInstanceManager::mutex_ = {};
+
+bool DsoInstanceManager::pub(std::string dso_id, std::string dso_name) {
+  if (getDsoInstanceByID(dso_id) != nullptr) {
+    ENVOY_LOG_MISC(warn, "pub {} {} dso instance failed: already pub.", dso_id, dso_name);
```

Comment:
What does this mean? Should this be an assert?

Reply:
Entering this branch indicates that two go extensions use the same `DSO`, for example, different listener blocks. I will adjust the log level and content. Thanks.

Reply:
Done